test-result-summary-client/src/Build/AllTestsInfo.jsx

```diff
@@ -17,7 +17,15 @@ export default class Build extends Component {
 
     async updateData() {
         const { buildId } = getParams( this.props.location.search );
-        const fetchBuild = await fetch( `/api/getAllTestsWithHistory?buildId=${buildId}&limit=5 `, {
+        const fetchThisBuild = await fetch(`/api/getData?_id=${buildId}`, {
+            method: 'get'
+        })
+        const build = await fetchThisBuild.json();
+        let limitNum = 5;
+        if ( build[0].type === "Perf" ) {
+            limitNum = 1;
+        }
```

Comment:
An additional query to figure out the type is not necessary. When calling `allTestsInfo ` (link below), we know the type, we can set the limit based on type and pass the limit as params.

https://github.com/AdoptOpenJDK/openjdk-test-tools/blob/cd800edf8d4b392d033611c4194d2707bd12acfb/test-result-summary-client/src/Build/BuildTable.jsx#L52

In AllTestsInfo.jsx, we can get the limit from the line 19 as following:
``` 
const { buildId, limit } = getParams( this.props.location.search );
```